File: src/main/groovy/net/zomis/brainf/model/BrainF.java
package net.zomis.brainf.model;

import net.zomis.brainf.model.ast.Lexer;
import net.zomis.brainf.model.ast.tree.Parser;
import net.zomis.brainf.model.groovy.GroovyBFContext;
import net.zomis.brainf.model.input.QueueInput;

import java.util.concurrent.BlockingQueue;

public class BrainF {
	public static BrainfuckRunner createUsingQueueWithMemorySize(BlockingQueue<Integer> input,
             int memorySize, BrainfuckOutput output) {
		return new BrainfuckRunner(new BrainfuckMemory(memorySize), new BrainfuckCode(), new QueueInput(input), output);
	}

    public static BrainfuckCode code(String code) {
        BrainfuckCode bfCode = new BrainfuckCode();
        bfCode.setRootTree(new Parser(new GroovyBFContext()).parse(Lexer.tokenize(code)));
        bfCode.setSource(ListCode.create(code));
        return bfCode;
    }
}

File: brainduck-kotlin/src/commonMain/kotlin/net/zomis/brainf/model/BrainF.kt
package net.zomis.brainf.model;

import net.zomis.brainf.model.ast.Lexer;
import net.zomis.brainf.model.ast.tree.Parser;
import net.zomis.brainf.model.classic.BrainfuckConverter
import net.zomis.brainf.model.groovy.GroovyBFContext;
import net.zomis.brainf.model.input.QueueInput

object BrainF {
	fun createUsingQueueWithMemorySize(
        input: Iterator<Int>,
        memorySize: Int,
        output: BrainfuckOutput,
    ): BrainfuckRunner {
		return BrainfuckRunner(BrainfuckMemory(memorySize), BrainfuckCode(), QueueInput(input), output)
	}

    fun code(code: String): BrainfuckCode {
        val bfCode = BrainfuckCode()
        val tokens = Lexer.tokenize(code)
        bfCode.setRootTree(Parser(GroovyBFContext()).parse(tokens))
        bfCode.source = ListCode.create(BrainfuckConverter(), code)
        return bfCode
    }
}
